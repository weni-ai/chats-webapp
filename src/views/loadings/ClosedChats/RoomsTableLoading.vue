<template>
  <section class="rooms-table-loading">
    <header class="rooms-table-loading__categories">
      <unnnic-skeleton-loading />
      <unnnic-skeleton-loading />
      <unnnic-skeleton-loading />
      <unnnic-skeleton-loading />
      <unnnic-skeleton-loading />
    </header>
    <section class="rooms-table-loading__row" v-for="row in 4" :key="`row-${row}`">
      <div class="rooms-table-loading__contact-info">
        <unnnic-skeleton-loading height="40px" />
        <unnnic-skeleton-loading />
      </div>
      <unnnic-skeleton-loading class="rooms-table-loading__agent" />
      <div class="rooms-table-loading__tags">
        <unnnic-skeleton-loading />
        <unnnic-skeleton-loading />
        <unnnic-skeleton-loading />
      </div>
      <unnnic-skeleton-loading class="rooms-table-loading__date" />
      <unnnic-skeleton-loading class="rooms-table-loading__show" />
    </section>
  </section>
</template>

<script>
export default {
  name: 'RoomsTableFiltersLoading',
};
</script>

<style lang="scss" scoped>
.rooms-table-loading {
  display: flex;
  flex-direction: column;

  overflow: hidden auto;

  &__categories,
  &__row {
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    gap: $unnnic-spacing-sm;
  }

  &__row {
    display: grid;
    align-items: center;

    padding: 12px 16px;
  }

  &__categories {
    padding: $unnnic-spacing-ant $unnnic-spacing-sm;
    margin-bottom: $unnnic-spacing-nano;

    & > * {
      height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    }
  }

  &__contact-info {
    display: grid;
    grid-template-columns: 40px auto;
    gap: $unnnic-spacing-nano;
    align-items: center;
  }

  &__tags {
    display: grid;
    grid-template-columns: 2fr 1fr 2fr;
    gap: $unnnic-spacing-xs;

    & > * {
      height: 40px;
    }
  }

  &__show {
    height: 38px;
  }
}
</style>
