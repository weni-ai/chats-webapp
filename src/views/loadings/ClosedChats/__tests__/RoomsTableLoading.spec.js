import { mount } from '@vue/test-utils';
import RoomsTableLoading from '../RoomsTableLoading.vue';
import { beforeEach, describe, expect } from 'vitest';

const createWrapper = () => {
  return mount(RoomsTableLoading);
};

describe('RoomsTableLoading.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
