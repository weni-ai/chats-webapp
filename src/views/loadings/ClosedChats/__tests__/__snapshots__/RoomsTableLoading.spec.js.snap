// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoomsTableLoading.vue > to match snapshot 1`] = `
"<section data-v-69f1dee3="" class="rooms-table-loading">
  <header data-v-69f1dee3="" class="rooms-table-loading__categories"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></header>
  <section data-v-69f1dee3="" class="rooms-table-loading__row">
    <div data-v-69f1dee3="" class="rooms-table-loading__contact-info"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__agent"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
    <div data-v-69f1dee3="" class="rooms-table-loading__tags"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__date"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3="" class="rooms-table-loading__show"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
  </section>
  <section data-v-69f1dee3="" class="rooms-table-loading__row">
    <div data-v-69f1dee3="" class="rooms-table-loading__contact-info"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__agent"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
    <div data-v-69f1dee3="" class="rooms-table-loading__tags"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__date"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3="" class="rooms-table-loading__show"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
  </section>
  <section data-v-69f1dee3="" class="rooms-table-loading__row">
    <div data-v-69f1dee3="" class="rooms-table-loading__contact-info"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__agent"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
    <div data-v-69f1dee3="" class="rooms-table-loading__tags"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__date"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3="" class="rooms-table-loading__show"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
  </section>
  <section data-v-69f1dee3="" class="rooms-table-loading__row">
    <div data-v-69f1dee3="" class="rooms-table-loading__contact-info"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__agent"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
    <div data-v-69f1dee3="" class="rooms-table-loading__tags"><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div><span data-v-69f1dee3="" class="rooms-table-loading__date"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-69f1dee3="" class="rooms-table-loading__show"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
  </section>
</section>"
`;
